js/script.js

```diff
@@ -0,0 +1,55 @@
+var userInput = "";
+var pictures = [];
+var i = 0;
+
+
+
+
+$(document).ready(function() {
+
+	$("form").on("submit", function (e) {
+		userInput = "";
+		i = 0;
+		pictures = [];
+		$("div.carousel-inner").html('');
+		$("ol.carousel-indicators").html('');
+		userInput = $("input").val();
+		$("input").val('')
+		e.preventDefault();
+		$.get("http://www.reddit.com/search.json", { q: userInput }).done(function(data) {
+			//get images out of array of objects
+			data.data.children.forEach(function(article) {
+				//filter out posts without thumbnails
+				if (article.data.thumbnail !== "self") {
+					if (article.data.preview) {
+						pictures.push(article.data.preview.images[0].source.url);
+					} else if (article.data.media) {
+						pictures.push(article.data.media.oembed.thumbnail_url);
+					} else if (article.data.url.indexOf(".jpg") != -1) {
+						pictures.push(article.data.url);
+					} else if (article.data.url.indexOf(".png") != -1) {
+						pictures.push(article.data.url);
+					} else if (article.data.url.indexOf(".gif") != -1) {
+						pictures.push(article.data.url);
+					}
+				}
+			});
```

Comment:
Nice job filtering out the images! See how you could do the same process with the `.map()` function.